### PR TITLE
imsolaris: rewrite documentation

### PR DESCRIPTION
**Pull request by PascalWithopf**
Module documentation was rewritten in the new style.

### CODE REVIEW

**Comment by deoren** [comment]
Is this another module that hasn't been converted to use the [`advanced`](http://www.rsyslog.com/doc/v8-stable/configuration/conf_formats.html) configuration format?

File: source/configuration/modules/imsolaris.rst [hunk 1]
```diff
@@ -12,30 +22,38 @@ syslogd instance. As such, only one can be active at one time. If it
...
-------------------------
+========================
+
+.. note::
+
+   Parameter names are case-insensitive.
 
-Note: parameter names are case-insensitive.
+
+Legacy Directives
```
Comment (deoren) [inline source/configuration/modules/imsolaris.rst]: Perhaps use the substitution reference here?

**Comment by deoren** [comment]
I found some issues with the recent changes to imrelp.rst, impstats.rst and imptcp.rst:

```
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/impstats.rst:291: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imrelp.rst
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imptcp.rst:540: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/impstats.rst
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imrelp.rst:343: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imptcp.rst
```

The conflict was not immediately obvious from CI since each PR stood alone and was not rebased against master between each merge (as a result the conflict was not seen).

This will require updating all module docs using the new format to use a unique label for the stats counter section.

I'll submit a PR for those module docs and imuxsock.rst soon.

**Comment by deoren** [comment]
> @deoren: This will require updating all module docs using the new format to use a unique label for the stats counter section.

Turns out that only the module docs already merged were affected.

**Comment by deoren** [comment]
@PascalWithopf Looks like I must have missed your recent commit (ef37344). Are you ready to have this merged?

**Comment by PascalWithopf** [comment]
@deoren Yes

**Comment by deoren** [comment]
@PascalWithopf Thanks.

**Pull request merged**